### PR TITLE
WIP: BLD: fix Azure CI issue related to relpath on different drive letters

### LINKED ISSUE #11967: BLD: Azure windows runs complain about drives
xref numpy/numpy#16101 where the fix there should be easy to port to `azure-pipelines.yml`

With that, I don't understand what is going on here. There are 4 windows runs: CPython36-32bit, CPython36, CPython37, CPython38. Only the middle two are failing with the "ValueError: path is on mount 'c:', start on mount 'D:'" message. In all the cases, Python is on drive C and the code is on drive D. There is a [warning](https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=5971&view=logs&j=9da2eb47-aaf8-5d70-4df4-40cdaefcc6f3&t=3b8c79d2-9141-5f8e-2c5d-55a8b0d38b7c&l=7476)
```
... numpy\distutils\system_info.py:820: UserWarning: \
Specified path c:\hostedtoolcache\windows\python\3.6.8\x64\include\include is invalid.
```

and then the error, which seems to come from trying to link with mingw fortran on files that are located on C. I do see that you are using `pip wheel --no-build-isolation -v -v -v --wheel-dir=dist .`, perhaps adding `--no-use-pep517` might change things, at least unless someone has already carefully checked that the pep517 build changes did not cause any problems (on numpy and python3.8 there is something wrong with the way the setup.py is constructed, the line `run_build = parse_setuppy_commands()` was erroring, and the error was ignored).

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-11967, see also https://github.com/numpy/numpy/issues/12530

related: gh-11965

### CODE REVIEW

**Comment by rgommers** [comment]
Hmm, that got us past building `fft`, but it still fails further on:
```
2020-05-01T11:40:55.1974685Z   building 'scipy.fft._pocketfft.pypocketfft' extension
2020-05-01T11:40:55.1976553Z   compiling C sources
2020-05-01T11:40:55.1977819Z   creating build\temp.win-amd64-3.6\Release\scipy\fft
2020-05-01T11:40:55.1979112Z   creating build\temp.win-amd64-3.6\Release\scipy\fft\_pocketfft
2020-05-01T11:40:55.1988833Z   C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\pybind11\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\pybind11\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\numpy\core\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt /EHsc /Tpscipy\fft\_pocketfft\pypocketfft.cxx /Fobuild\temp.win-amd64-3.6\Release\scipy\fft\_pocketfft\pypocketfft.obj /std:c++14 /EHsc /Zm1000
2020-05-01T11:41:00.3456858Z   C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\ProgramData\chocolatey\lib\mingw\tools\install\mingw64\lib\gcc\x86_64-w64-mingw32\6.4.0 /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\libs /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\PCbuild\amd64 /LIBPATH:build\temp.win-amd64-3.6 /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\libs /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\PCbuild\amd64 /LIBPATH:build\temp.win-amd64-3.6 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\lib\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\lib\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x64 /EXPORT:PyInit_pypocketfft build\temp.win-amd64-3.6\Release\scipy\fft\_pocketfft\pypocketfft.obj /OUT:build\lib.win-amd64-3.6\scipy\fft\_pocketfft\pypocketfft.cp36-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.6\Release\scipy\fft\_pocketfft\pypocketfft.cp36-win_amd64.lib
2020-05-01T11:41:16.1142037Z   building 'scipy.fftpack.convolve' extension
2020-05-01T11:41:16.1143617Z   compiling C sources
2020-05-01T11:41:16.1144584Z   creating build\temp.win-amd64-3.6\Release\scipy\fftpack
2020-05-01T11:41:16.1155160Z   C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\numpy\core\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt /Tcscipy\fftpack\convolve.c /Fobuild\temp.win-amd64-3.6\Release\scipy\fftpack\convolve.obj
2020-05-01T11:41:16.3692808Z   C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\ProgramData\chocolatey\lib\mingw\tools\install\mingw64\lib\gcc\x86_64-w64-mingw32\6.4.0 /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\libs /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\PCbuild\amd64 /LIBPATH:build\temp.win-amd64-3.6 /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\libs /LIBPATH:c:\hostedtoolcache\windows\python\3.6.8\x64\PCbuild\amd64 /LIBPATH:build\temp.win-amd64-3.6 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\lib\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\lib\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x64 /EXPORT:PyInit_convolve build\temp.win-amd64-3.6\Release\scipy\fftpack\convolve.obj /OUT:build\lib.win-amd64-3.6\scipy\fftpack\convolve.cp36-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.6\Release\scipy\fftpack\convolve.cp36-win_amd64.lib
2020-05-01T11:41:17.7987299Z   building 'scipy.integrate._quadpack' extension
2020-05-01T11:41:17.7988768Z   compiling C sources
2020-05-01T11:41:17.7990895Z   creating build\temp.win-amd64-3.6\Release\scipy\integrate
2020-05-01T11:41:17.8000204Z   C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DHAVE_CBLAS -Iscipy\_lib\src -Ic:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\numpy\core\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\include -Ic:\hostedtoolcache\windows\python\3.6.8\x64\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt /Tcscipy\integrate\_quadpackmodule.c /Fobuild\temp.win-amd64-3.6\Release\scipy\integrate\_quadpackmodule.obj
2020-05-01T11:41:17.9230283Z   could not find library 'quadpack' in directories ['D:\\a\\1\\s\\build\\openblas_lapack_info']
2020-05-01T11:41:17.9231580Z   could not find library 'mach' in directories ['D:\\a\\1\\s\\build\\openblas_lapack_info']
2020-05-01T11:41:17.9232373Z   could not find library 'openblas_lapack_info' in directories ['D:\\a\\1\\s\\build\\openblas_lapack_info']
```

**Comment by rgommers** [comment]
Adding the relevant `system_info` output:
``
2020-05-01T11:38:19.8126877Z       libraries openblas not found in ['c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib', 'C:\\', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\libs']
2020-05-01T11:38:19.8130505Z     Library openblas was not found. Ignoring
2020-05-01T11:38:19.8131532Z       FOUND:
2020-05-01T11:38:19.8132352Z         library_dirs = ['D:\\a\\1\\s\\build\\openblas_lapack_info']
2020-05-01T11:38:19.8133172Z         libraries = ['openblas_lapack_info']
2020-05-01T11:38:19.8134006Z         language = f77
2020-05-01T11:38:19.8134772Z         define_macros = [('HAVE_CBLAS', None)]
2020-05-01T11:38:19.8135465Z 
2020-05-01T11:38:19.8137217Z       FOUND:
2020-05-01T11:38:19.8138292Z         library_dirs = ['D:\\a\\1\\s\\build\\openblas_lapack_info']
2020-05-01T11:38:19.8140010Z         libraries = ['openblas_lapack_info']
2020-05-01T11:38:19.8140977Z         language = f77
```

**Comment by rgommers** [comment]
Ah wait, there are two independent bugs here, `ValueError: path is on mount 'c:', start on mount 'D:'` is still present on the `ilp64` build.

Having `openblas_lapack_info` appear in `library_dirs` is a bug in `system_info.py`.

**Comment by tylerjereddy** [comment]
I was looking at this a bit in my [fork](https://github.com/tylerjereddy/scipy/pull/31) too. I tried symlinking the mingw `bin` dir to a path on the same drive letter/working dir as the scipy clone. 

```
      cmd /c mklink /d "$pwd\\bin" "C:\\ProgramData\\chocolatey\\lib\\mingw\\tools\\install\\mingw$(BITS)\\bin" 
      $env:PATH = "$pwd\\bin;" + $env:PATH
```

It changed the error to a low-level gfortran/assembly issue:

```
 gfortran.exe:f77: scipy\integrate\quadpack\dqagpe.f
  C:\Users\VSSADM~1\AppData\Local\Temp\ccKmDIZs.s: Assembler messages:
  C:\Users\VSSADM~1\AppData\Local\Temp\ccKmDIZs.s:20: Error: unknown .loc sub-directive `view'
  C:\Users\VSSADM~1\AppData\Local\Temp\ccKmDIZs.s:20: Error: junk at end of line, first unrecognized character is `-'
```

Perhaps I needed more than the `bin` dir linked over (`lib`/`include` dirs too?). Could also try recursive copying. Cleaner would be to just install it in the right place, but that's apparently a ["pro"](https://stackoverflow.com/a/19777121/2942522) feature for `choco` package manager.

Anyway, not sure this approach is any more productive, I may play with it a bit..

**Comment by pv** [comment]
> Having openblas_lapack_info appear in library_dirs is a bug in system_info.py.

The "fake" library objects have to named uniquely for each of the system_info classes, and it just uses the class name for simplicity (even if maybe looks confusing in the long). So I think it's not a bug.

File: setup.py [hunk 1]
```diff
@@ -278,6 +278,48 @@ def __is_using_gnu_linker(self, ext):
                     is_gcc = "gcc" in compiler_name or "g++" in compiler_name
             return is_gcc and sysconfig.get_config_var('GNULD') == 'yes'
 
+    # Overriding due to https://github.com/numpy/numpy/issues/12530, which
+    # is showing up in Azure CI as:
+    #         ValueError: path is on mount 'c:', start on mount 'D:'
+    def _process_unlinkable_fobjects(self, objects, libraries,
```
Comment (pv) [inline setup.py:284]: Note that this is just function definition (see the indentation)... it doesn't actually override anything

**Comment by tylerjereddy** [comment]
Moving the mingw install to the D: root path worked eventually, but when I tried to move openblas64_.a to the same root path distutils couldn't find it even with a site.cfg:

```
[openblas64_]
libraries = openblas64_
library_dirs = /d/a/1/s/openblaslib
include_dirs = /d/a/1/s/openblaslib
runtime_library_dirs = /d/a/1/s/openblaslib
```

Some of those entries are useless for static library I suppose. I don't know if the pats get normalized across platforms either, though I'd imagine they might.

**Comment by pv** [comment]
Is that `/d/...` the mingw/cygwin drive syntax? Python reads that file, and I'm not sure it will understand that `/d/` means `d:\`

**Comment by tylerjereddy** [comment]
Same result with "genuine" paths:

```
 68   - bash: |
 69       echo "[openblas64_]" >> site.cfg                                                                                                                    
 70       echo "libraries = openblas64_" >> site.cfg                                                                                                          
 71       echo "library_dirs = d:\a\1\s\openblaslib" >> site.cfg                                                                                              
 72       echo "include_dirs = d:\a\1\s\openblaslib" >> site.cfg                                                                                              
 73       echo "runtime_library_dirs = d:\a\1\s\openblaslib" >> site.cfg                                                                                      
 74       cat site.cfg
 75     displayName: 'Set site.cfg for ilp64 case'                                                                                                            
 76     condition: eq(variables['NPY_USE_BLAS_ILP64'], 1)          
```

**Comment by pv** [comment]
Do you have a build log?

**Comment by tylerjereddy** [comment]
Yup, maybe the  thing you mentioned about needing the 32-bit integer static lib as well?

https://dev.azure.com/tylerjereddy/tyler-scipy-fork/_build/results?buildId=1024&view=logs&j=da2f0622-108b-5827-a4af-f8cb335e1cfa

Would a second `site.cfg` entry be needed for plain `openblas` then?

**Comment by pv** [comment]
Yes, adding also the `[openblas]` section hopefully makes it to work...

**Comment by tylerjereddy** [comment]
CI all green in master now after cross-ref'd PR above was merged. We'll have more refinements to come in the future of course, esp. if the upstream handling improves.

Thanks Ralf/Pauli

**Comment by rgommers** [comment]
Thanks @tylerjereddy and @pv.

**Pull request closed without merging**